web/src/lib/utils/folder-utils.ts

```diff
@@ -0,0 +1,18 @@
+export interface RecursiveObject {
+  [key: string]: RecursiveObject;
+}
+
+export function buildFolderTree(paths: string[]) {
+  const root: RecursiveObject = {};
+  for (const path of paths) {
+    const parts = path.split('/');
+    let current = root;
+    for (const part of parts) {
+      if (!current[part]) {
```

Comment:
`!{}` is also true, which makes this a bit weird imo. I think we should be more explicit here by writing `current[part] === undefined`

Reply:
```
> !{}
false
```

Reply:
Oh huh, that's weird.